src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
+    public void checkPositionIndex(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index is invalid");
+        }
+    }
+
+    public void checkElementIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index is invalid");
+        }
+    }
+
+    void linkLast(T t) {
```

Comment:
Change local variable names. They must be more informative. Replace 't' to "value" here and another place. 
```suggestion
    void linkLast(T value) {
```